### PR TITLE
Updating attribute calculation to look at all equipment #8188

### LINKED ISSUE #8185: Back and Body Accessory battle gear don't add to user's equipment stats / attributes
When you equip a Back Accessory or Body Accessory as Battle Gear, your attributes aren't boosted by the item's stats (see screenshot below).

_[Guesswork in this paragraph.]_ This is presumably because stats-boosting accessories have only recently been created and the code for calculating and displaying attributes might not take those item types into account. If so, a fix is probably simple: replicate what's been done for the other equipment types. However this might be an opportunity to improve the code to avoid it handling each equipment type individually.

image

### PR DESCRIPTION
**Pull request by Hus274**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #8185 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
I've refactored the code around stat calculation to try and reduce duplicate logic to calculate stat totals and specific components of stat calculations.

I've also updated the attribute calculation to include all equipped items instead of just specific sections of it.

screen shot 2016-11-09 at 5 28 45 pm

### CODE REVIEW

File: test/common/fns/statsComputed.test.js [hunk 1]
```diff
@@ -25,4 +25,71 @@ describe('common.fns.statsComputed', () => {
...
+  it('calculates calculates stat bonuses for class', () => {
+    user.items.gear.equipped.weapon = 'weapon_warrior_1';
+    let result = statsComputed(user);
+
+    expect(result.str).to.eql(4.5);
+    expect(result.gearBonus.str).to.eql(3);
+    expect(result.classBonus.str).to.eql(1.5);
+  });
+
+  it('calculates calculates stat bonuses for level', () => {
```
Comment (khipkin) [inline test/common/fns/statsComputed.test.js]: "calculates calculates" a few times in this file

File: test/common/fns/statsComputed.test.js [hunk 2]
```diff
@@ -25,4 +25,71 @@ describe('common.fns.statsComputed', () => {
...
+
+  it('sets buffs field', () => {
+    user.stats.buffs.str = 150;
+    let result = statsComputed(user);
+
+    expect(result.str).to.eql(150);
+    expect(result.buff.str).to.eql(150);
+  });
+
+  it('calculate mp from intelligent', () => {
```
Comment (khipkin) [inline test/common/fns/statsComputed.test.js]: intelligence

**Comment by Hus274** [comment]
fixed spellings.

**Comment by Hus274** [comment]
It looks like [this commit](https://github.com/HabitRPG/habitica/commit/eaaab35f314e46c81e2c042db5dc1a28daa13275) was just merged and added the computation for the back and body pieces of equipment but simply added them to the array. Do we still want this change to make it more generic? If so, I'll work on fixing the conflicts in this branch.

**Comment by Hus274** [comment]
Sorry, just saw the comment on the original issue. I'll resolve the conflicts here tonight and push a new version.

**Comment by Hus274** [comment]
I've merged this back into the mainline and removed all of the conflicts associated with the commit I referenced earlier. This should be good to go now, let me know if anything else is needed here!

**Comment by TheHollidayInn** [comment]
Noted towards your Tier! Thanks :D

**Pull request merged**